### PR TITLE
GCC compiler error fixes and version update

### PR DESCRIPTION
**Pull request by tdjustus**
Version 0.4.2

### CODE REVIEW

File: src/render.c [hunk 1]
```diff
@@ -91,7 +91,7 @@ void render_init (Render* render, Args* args)
 	float scale = MIN(
 			display->screen->w/(float)BASE_W,
 			display->screen->h/(float)BASE_H
-		) * args->antialiasing ? 2 : 1;
+		) && args->antialiasing ? 2 : 1;
```
Comment (bart9h) [inline src/render.c:94]: This seem wrong. I can't see how it could work. What was the compiler error you were trying to fix here?

**Review by bart9h — CHANGES_REQUESTED** [review]
some changes are wrong, others are pointless

File: src/render.c (hunk 1 above)
Comment (tdjustus) [reply src/render.c:94]: make -C src
make[1]: Entering directory '/home/timothy/cave9/src'
cc -std=c99 -Wall -Werror -ggdb `sdl-config --cflags` -I/usr/local/include/SDL -I/usr/include/SDL   -c -o main.o main.c
cc -std=c99 -Wall -Werror -ggdb `sdl-config --cflags` -I/usr/local/include/SDL -I/usr/include/SDL   -c -o audio.o audio.c
cc -std=c99 -Wall -Werror -ggdb `sdl-config --cflags` -I/usr/local/include/SDL -I/usr/include/SDL   -c -o control.o control.c
cc -std=c99 -Wall -Werror -ggdb `sdl-config --cflags` -I/usr/local/include/SDL -I/usr/include/SDL   -c -o display.o display.c
cc -std=c99 -Wall -Werror -ggdb `sdl-config --cflags` -I/usr/local/include/SDL -I/usr/include/SDL   -c -o game.o game.c
cc -std=c99 -Wall -Werror -ggdb `sdl-config --cflags` -I/usr/local/include/SDL -I/usr/include/SDL   -c -o render.o render.c
render.c: In function ‘render_init’:
render.c:94:19: error: ‘*’ in boolean context, suggest ‘&&’ instead [-Werror=int-in-bool-context]
   94 |                 ) * args->antialiasing ? 2 : 1;
render.c: In function ‘render_hud’:
render.c:416:53: error: ‘%s’ directive output may be truncated writing up to 127 bytes into a region of size 122 [-Werror=format-truncation=]
  416 |                 snprintf (buf, HUD_TEXT_MAX, "score %s", score);
      |                                                     ^~   ~~~~~
render.c:416:17: note: ‘snprintf’ output between 7 and 134 bytes into a destination of size 128
  416 |                 snprintf (buf, HUD_TEXT_MAX, "score %s", score);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[1]: *** [<builtin>: render.o] Error 1
make[1]: Leaving directory '/home/timothy/cave9/src'
make: *** [GNUmakefile:17: build] Error 2

Comment (bart9h) [reply src/render.c:94]: I see. But you shouldn't follow compiler suggestions blindly. The render.c:94 "error" should be solved with a parenthesis instead, like `* (args->antialiasing ? 2 : 1)`.  Using `&&` makes no sense in this context.

Comment (bart9h) [reply src/render.c:94]: I fixed the errors myself. Thanks for bringing them to my attention. I haven't compiled cave9 for years, and GCC is more pedantic nowadays.

**Comment by bart9h** [comment]
the compilation errors were fixed

**Pull request closed without merging**